app/(onboard)/_components/OnboardSteps/OnboardSteps.tsx

```diff
@@ -0,0 +1,61 @@
+'use client';
+
+import { Check } from 'lucide-react';
+
+interface OnboardStepsProps {
+  currentStep: string;
+}
+
+function OnboardSteps({ currentStep }: OnboardStepsProps) {
+  const steps = [
+    {
+      number: 1,
+      description: 'Create Account',
+      optional: false,
+    },
+    {
+      number: 2,
+      description: 'Configure Study',
+      optional: true,
+    },
+    {
+      number: 3,
+      description: 'Documentation',
+      optional: false,
+    },
+  ];
```

Comment:
I don't love this for a few reasons:

- Seems like we have two sources of truth for the steps. There's this, and then there's `OnboardWizard`. A better design would be to have a single definition of the steps, including the component to render for that step. That way, `OnboardSteps` could render the description, and `OnboardWizard` could render the component.
- Relatedly, I think we should be defining a `Step` type, which reflects the properties it has.
- I don't like the parsing as Int happening on line 28. It feels really weird to pass a prop into a component, create a type definition that says its a string, and then immediately convert it into an integer. We should be passing components props in the format they are expecting, as a general principle. Exceptions would be where a component does an enormous amount of wrangling of data (such as perhaps rendering a data view, where the raw input is a database query output). the reason for this I can see is that this is coming from queryParams which are cast as a string by nextJS. So I'll leave some notes about fixing it there.

Reply:
all resolved with https://github.com/complexdatacollective/Fresco/pull/10/commits/958fc128ece438207310ff14e8076082b8f7b62a